session.go

```diff
@@ -0,0 +1,160 @@
+package mockoidc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// TimeFunc is the default function that the token generators will use for the current time
+var TimeFunc = time.Now
+
+// TimeNowFunc is the type of the function that can be passed tot he topken generators
+// if time.Now needs to be overriden for testing purposes
+type TimeNowFunc func() time.Time
+
+// Session is a struct that maintains session state across requests
+type Session struct {
+	SessionID  string
+	Scopes     []string
+	OAuthState string
+	OIDCNonce  string
+	User       *User
+}
+
+// SessionStore is a map of #Session structs
+type SessionStore struct {
+	Store map[string]*Session
+}
+
+// Config is the OIDCMock server configuration information (here as placeholder)
+type Config struct {
+	ClientID     string
+	ClientSecret string
+	Keypair      *Keypair
+	AccessTTL    int64
+	RefreshTTL   int64
+	Issuer       string
+}
+
+// NewSession returns a pointer to a new Session struct
+func (ss *SessionStore) NewSession(scope string, oAuthState string, oidcNonce string, user *User) (*Session, error) {
+	sessionID, err := nonce(24)
+	if err != nil {
+		return nil, err
+	}
+
+	session := &Session{
+		SessionID:  sessionID,
+		Scopes:     strings.Split(scope, " "),
+		OAuthState: oAuthState,
+		OIDCNonce:  oidcNonce,
+		User:       user,
+	}
+
+	ss.Store[sessionID] = session
+
+	return session, nil
+}
+
+// AccessToken returns the JWT token with the appropriate claims for an access token
+func (s *Session) AccessToken(config *Config, timeFunc TimeNowFunc) (string, error) {
+
+	claims := standardClaims(s, config, config.AccessTTL, timeFunc)
+
+	token, err := config.Keypair.SignJWT(claims)
+	if err != nil {
+		return "", err
+	}
+
+	return token, nil
+}
+
+// standardClaims returns a populated jwt.StandardCLaims struct
+func standardClaims(s *Session, config *Config, ttl int64, timeFunc TimeNowFunc) jwt.StandardClaims {
```

Comment:
Lets move `s *Session` to be the receiver of this:
```suggestion
func (s *Session) standardClaims(config *Config, ttl int, now time.Time) jwt.StandardClaims {
```